core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseSampleSourceTelemetry.java

```diff
@@ -17,6 +17,8 @@ public void reset() {
     @Override
     protected void setSampleRate(Envelope envelope) {
         Double currentSP = getSamplingPercentage();
-        envelope.setSampleRate(currentSP);
+        if (currentSP != null) {
```

Comment:
I got your question now. I checked the implementation of reset and right now it tries to pass null and null gets set. I don't think so it is setting it back to 100 though I may be wrong

public void reset() {
        this.setSamplingPercentage((Double)null);
    }